.github/scripts/check_content.py

```diff
@@ -0,0 +1,57 @@
+import os.path
+from os import walk
+
+
+def check_content() -> int:
+    """
+    Walks through the filetree rooted at the current working directory.
+    For each file that it finds, it validates the file
+    """
+    error_count = 0
+    DIRECTORIES_TO_EXCLUDE = ['.git', '.github', '.idea', 'venv', 'DO NOT COMMIT']
```

Comment:
Some people (me) call their virtual environment `.venv`. How should we deal with all possibilities of virtual environment names? Maybe a addition (or creation if not yet exists) to a note on "scripts for the hub" explaining to add your virtual environment directory to this list?

Reply:
I'm not sure how to encode this feedback in the code, but might first thought was to add a comment before that code saying something like:

```py
# Note to future developers: if a development-related directory gets in the way here,
# feel free to add it to this list
```

On further thought, I think a better idea is to re-use the code in `MocFileAndDirectoryFilter` in `make_mocs.py` which has more sophisticated logic in it, like excluding everything - files and directories - that begins with a `.` - which would simplify the code here greatly.

Reply:
Whichever you decide to choose I'll agree with - just wanted to raise awareness. (feel free to resolve)

Reply:
> Some people (me) call their virtual environment `.venv`. How should we deal with all possibilities of virtual environment names? Maybe a addition (or creation if not yet exists) to a note on "scripts for the hub" explaining to add your virtual environment directory to this list?

@dbarenholz This was really helpful... It prompted me to make these two simplifications:

Re-use MocFileAndDirectoryFilter to filter directories
https://github.com/obsidian-community/obsidian-hub/pull/213/commits/a5c38734e607af8bf3f5af35c08c34b83d6264f1

Simplify the check for dot files in check_content.py
https://github.com/obsidian-community/obsidian-hub/pull/213/commits/0f4efb21a69eed88d5209a461255b2582ab74dea